### ISSUE TITLE
Sync slowdown

### ISSUE THREAD

**Issue by pepyakin** [paritytech/polkadot#611]
This is a meta issue to collect together all reports & information relating to massive slowdowns in the syncing.

---

Original issue:

I experienced massive slowdown running polkadot 0.6.17 (0929fe24), I launched my node to sync and it started running fast (although not native-fast) but the performance degraded really quick to around 2bps

```
2019-11-24 12:52:03 Discovered new external address for our node: /ip4/10.0.1.177/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG
2019-11-24 12:52:04 Discovered new external address for our node: /ip4/10.0.1.30/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG                                                               2019-11-24 12:52:05 Discovered new external address for our node: /ip4/89.245.190.29/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG
2019-11-24 12:52:07 Syncing, target=#805602 (5 peers), best: #649017 (0x6d9c…3319), finalized #648649 (0xca0c…369b), ⬇ 61.5kiB/s ⬆ 4.7kiB/s
2019-11-24 12:52:12 Syncing  0.0 bps, target=#805602 (8 peers), best: #649017 (0x6d9c…3319), finalized #648649 (0xca0c…369b), ⬇ 185.9kiB/s ⬆ 9.5kiB/s
2019-11-24 12:52:17 Syncing 60.6 bps, target=#805603 (8 peers), best: #649320 (0x84d1…b071), finalized #649195 (0xb097…393d), ⬇ 36.9kiB/s ⬆ 18.5kiB/s
2019-11-24 12:52:20 New epoch 289 launching at block 0xf00b…0a66 (block slot 262275439 >= start slot 262275439).                                                                                            2019-11-24 12:52:20 Next epoch starts at slot 262277839                                                                                                                                                     2019-11-24 12:52:20 Applying authority set change scheduled at block #650006
2019-11-24 12:52:20 Applying GRANDPA set change to new set with 100 authorities                                                                                                                             2019-11-24 12:52:20 Imported justification for block #650006 that triggers command Changing authorities, signaling voter.
2019-11-24 12:52:22 Syncing 212.0 bps, target=#805604 (14 peers), best: #650381 (0x3564…3143), finalized #650240 (0x4d13…ce6b), ⬇ 142.8kiB/s ⬆ 31.9kiB/s
2019-11-24 12:52:27 Syncing 156.9 bps, target=#805605 (17 peers), best: #651166 (0x675d…c31b), finalized #650752 (0x438c…0638), ⬇ 70.4kiB/s ⬆ 30.5kiB/s
2019-11-24 12:52:27 Discovered new external address for our node: /ip4/10.10.10.1/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG
2019-11-24 12:52:29 Discovered new external address for our node: /ip4/10.0.3.9/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG
2019-11-24 12:52:32 Syncing 122.9 bps, target=#805606 (19 peers), best: #651781 (0xd250…ae61), finalized #651776 (0x5d24…0f1c), ⬇ 241.1kiB/s ⬆ 22.4kiB/s
2019-11-24 12:52:33 Discovered new external address for our node: /ip4/192.168.1.65/tcp/30333/p2p/QmaiBVEdqxPSYxnMXLRhNroPYknuJ1MWcTBd32ftPJpCYG
2019-11-24 12:52:36 New epoch 290 launching at block 0x1079…df16 (block slot 262277840 >= start slot 262277839).                                                                                            2019-11-24 12:52:36 Next epoch starts at slot 262280239                                                                                                                                                     2019-11-24 12:52:36 Applying authority set change scheduled at block #652220
2019-11-24 12:52:36 Applying GRANDPA set change to new set with 100 authorities
2019-11-24 12:52:36 Imported justification for block #652220 that triggers command Changing authorities, signaling voter.
2019-11-24 12:52:37 Syncing 110.1 bps, target=#805607 (20 peers), best: #652332 (0x8d82…fe2e), finalized #652288 (0xe48e…b616), ⬇ 78.7kiB/s ⬆ 20.9kiB/s
2019-11-24 12:52:42 Syncing 100.9 bps, target=#805607 (21 peers), best: #652837 (0x3533…b9c6), finalized #652800 (0x70b1…a169), ⬇ 73.1kiB/s ⬆ 13.1kiB/s
2019-11-24 12:52:47 Syncing 90.3 bps, target=#805608 (22 peers), best: #653289 (0x7bbf…9cfd), finalized #652800 (0x70b1…a169), ⬇ 67.0kiB/s ⬆ 32.7kiB/s
2019-11-24 12:52:52 Syncing  7.9 bps, target=#805609 (23 peers), best: #653329 (0xe033…e61f), finalized #653312 (0xc010…f0db), ⬇ 97.9kiB/s ⬆ 10.7kiB/s
2019-11-24 12:52:57 Syncing  5.7 bps, target=#805610 (23 peers), best: #653358 (0xff64…ef48), finalized #653312 (0xc010…f0db), ⬇ 83.0kiB/s ⬆ 18.6kiB/s
```

after this it kept being at 2 bps.

After restarting the node, it got up to normal native speed at ~500bps.

I am conducting my tests on the beast. There is no memory pressure at all (39Gb used out of 256). There were no command line arguments except specifying the base dir.

**Comment by gavofyork** [comment]
Not much that can be done without further logs (`-lsync=trace`) and/or reproducability.

**Comment by pepyakin** [comment]
Also, I think somebody mentioned that it was related to rocksdb updates, so [this](https://github.com/paritytech/substrate/pull/4236) might have been fixed this.

**Comment by arkpar** [comment]
Right, this is fixed in 0.7.1 I believe
